src/server/ua_server_async.c

```diff
@@ -204,14 +205,10 @@ UA_AsyncManager_createAsyncResponse(UA_AsyncManager *am, UA_Server *server,
                                     const UA_AsyncOperationType operationType,
                                     UA_AsyncResponse **outAr) {
     UA_AsyncResponse *newentry = (UA_AsyncResponse*)UA_calloc(1, sizeof(UA_AsyncResponse));
-    if(!newentry)
-        return UA_STATUSCODE_BADOUTOFMEMORY;
+    UA_CHECK_MEM(newentry, return UA_STATUSCODE_BADOUTOFMEMORY);
 
     UA_StatusCode res = UA_NodeId_copy(sessionId, &newentry->sessionId);
-    if(res != UA_STATUSCODE_GOOD) {
-        UA_free(newentry);
-        return res;
-    }
+    UA_CHECK(res, goto error);
```

Comment:
goto is not evil.
But it is still harder to read.
Please put the free in the UA_Check macro as well.

Reply:
You can also review the file `ua_server.c`, it is also refactored with the use of check macros.

Okay it will be changed. Although there are some places in code where cleanup is done in a cleanup routine at the bottom of a function. Because I prever it that way I also modified the instance in this case. Maybe we find a rule of thumb when its is considered good to jump into a cleanup routine and when its not. (e.g. when there are at least two places in the same function that use the same cleanup routine).

You could review such an example in `ua_server.c` on lines 132 - 152.

Reply:
> Maybe we find a rule of thumb when its is considered good to jump into a cleanup routine and when its not. (e.g. when there are at least two places in the same function that use the same cleanup routine).

I like this rule!
Lets collect best practices in the CONTRIBUTING.md file.